src/main/java/ch/njol/skript/util/LiteralUtils.java

```diff
@@ -40,12 +40,19 @@ public class LiteralUtils {
 	 * @param <T>  {@code expr}'s type
 	 * @return {@code expr} without {@link UnparsedLiteral} objects
 	 */
-	@SuppressWarnings("unchecked")
+	@SuppressWarnings({"unchecked"})
 	public static <T> Expression<T> defendExpression(Expression<?> expr) {
 		if (expr instanceof ExpressionList) {
-			Expression<?>[] expressions = ((ExpressionList) expr).getExpressions();
-			for (int i = 0; i < expressions.length; i++)
-				expressions[i] = LiteralUtils.defendExpression(expressions[i]);
+			Expression<?>[] oldExpressions = ((ExpressionList<?>) expr).getExpressions();
+
+			Expression<? extends T>[] newExpressions = new Expression[oldExpressions.length];
```

Comment:
Is it possible that T would be different for the different expressions?

Reply:
Well the T doesn't mean much, it's just to remove the need for casting (in source):
https://github.com/SkriptLang/Skript/blob/1f5b4219b8d698e4548c8176f7a306b27f9ffdd4/src/main/java/ch/njol/skript/util/LiteralUtils.java#L44

And the expressions in an `ExpressionList<T>` are `Expression<? extends T>`